services/ux-backend/handlers/expandstorage/handler.go

```diff
@@ -0,0 +1,287 @@
+			"csi.storage.k8s.io/node-stage-secret-name":             "rook-csi-rbd-node",
+			"csi.storage.k8s.io/node-stage-secret-namespace":        namespace,
+			"csi.storage.k8s.io/controller-expand-secret-name":      "rook-csi-rbd-provisioner",
+			"csi.storage.k8s.io/controller-expand-secret-namespace": namespace,
+		},
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.Context(), client, storageClass, mutate)
+	if err != nil {
+		klog.Errorf("failed to create storageClass: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func createCephFilesystem(w http.ResponseWriter, r *http.Request, client client.Client, poolName, lsoStorageClassName string, mutate func() error, dataProtectionPolicy int, enableCompression bool, filesystemName, namespace string) {
+	compressiom := "none"
+	if enableCompression {
+		compressiom = "aggressive"
+	}
+	// get the current filesystem
+	cephFilesystem := &cephv1.CephFilesystem{}
+	err := client.Get(r.Context(), types.NamespacedName{Name: filesystemName, Namespace: namespace}, cephFilesystem)
+	if err != nil {
+		klog.Errorf("failed to get cephFilesystem: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	datapool := cephv1.NamedPoolSpec{
+		Name: poolName,
+		PoolSpec: cephv1.PoolSpec{
+			FailureDomain: "host",
+			DeviceClass:   lsoStorageClassName,
+			Replicated: cephv1.ReplicatedSpec{
+				Size:                   uint(dataProtectionPolicy),
+				RequireSafeReplicaSize: true,
+			},
+			Parameters: map[string]string{
+				"compression_mode": compressiom,
+			},
+		},
+	}
+
+	// update the filesystem with the new data pool
+	cephFilesystem.Spec.DataPools = append(cephFilesystem.Spec.DataPools, datapool)
+	_, err = ctrl.CreateOrUpdate(r.Context(), client, cephFilesystem, mutate)
+	if err != nil {
+		klog.Errorf("failed to update cephFilesystem: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func createCephFilesystemStorageClass(w http.ResponseWriter, r *http.Request, client client.Client, storageClassName, poolName, reclaimPolicy, volumeBindingMode string, enableEncryption bool, filesystemName, namespace string, mutate func() error) {
+	allowVolumeExpansion := true
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: storageClassName,
+			Annotations: map[string]string{
+				"description": "Provides RWO and RWX Filesystem volumes",
+			},
+		},
+		Provisioner:       storagecluster.CephFSDriverName,
+		ReclaimPolicy:     (*corev1.PersistentVolumeReclaimPolicy)(&reclaimPolicy),
+		VolumeBindingMode: (*storagev1.VolumeBindingMode)(&volumeBindingMode),
+		// AllowVolumeExpansion is set to true to enable expansion of OCS backed Volumes
+		AllowVolumeExpansion: &allowVolumeExpansion,
+		Parameters: map[string]string{
+			"clusterID":  namespace,
+			"fsName":     filesystemName,
+			"pool":       poolName,
+			"encryption": strconv.FormatBool(enableEncryption),
```

Comment:
just want to point out that UI doesn't support encryption for CephFS prov classes yet... I hope Eran is aware that this feature does not follow existing behaviour, if not then please bring this up with him once.

Reply:
then enable encryption checkbox should be hidden or disabled then filesystem is selected cc @vbnrh